tests/codegen/test_parity_native.py

```
@@ -0,0 +1,171 @@
+def _gen_module(dotted, prefix, base, name_pattern, parser_module):
+    eps = en.extract_module(dotted, prefix, base, name_pattern)
+    doc = {
+        "api": dotted.rsplit(".", 1)[-1],
+        "host": base,
+        "name_pattern": name_pattern,
+        "module": dotted.rsplit(".", 1)[-1],
+        "parser_module": parser_module,
+        "runtime_imports": ["_get", "format_nhl_season"],
+        "endpoints": eps,
+    }
+    yp = Path(tempfile.mktemp(suffix=".yaml"))
```

Ruff (0.15.15)

[error] Use of insecure and deprecated function (`mktemp`)

(S306)